### PR TITLE
docs(js): Give more emphasis on using autocomplete and instantsearch

### PR DESCRIPTION
**Pull request by pixelastic**
We would like people to use autocomplete and instantsearch instead of
building their UIs manually, so I'm giving a bit more emphasis on
those libraries in the doc.

I've added animated gif for explaining what autocomplete and what instantsearch is about. The gif are not included in this PR, but in another PR done on the javascript client (https://github.com/algolia/algoliasearch-client-js/pull/259)

### CODE REVIEW

File: include/templates/README.md [hunk 1]
```diff
@@ -175,16 +175,48 @@ Quick Start
...
+![autocomplete.js example](https://raw.githubusercontent.com/algolia/algoliasearch-client-js/master/examples/autocomplete.gif)
+
+#### [instantsearch.js](https://community.algolia.com/instantsearch.js/)
+
+[instantsearch.js](https://community.algolia.com/instantsearch.js/) is for **full
+page** search.
+
+![instantsearch.js example](https://raw.githubusercontent.com/algolia/algoliasearch-client-js/master/examples/instantsearch.gif)
+
+We strongly encourage you to have a look at this libraries because they are
```
Comment (vvo) [inline include/templates/README.md]: `those`?

File: include/templates/README.md [hunk 2]
```diff
@@ -175,16 +175,48 @@ Quick Start
 <% if js? -%>
 ### Frontend
 
+The JavaScript API client gives you access to low level methods to search and
+receive results. This is all you need for building your front-end but will
+require a bit of custom code on your side for displaying the results.  Reading
```
Comment (vvo) [inline include/templates/README.md]: extra space

**Comment by pixelastic** [comment]
@vvo: Fixed the extra space and typo

File: include/templates/README.md [hunk 3]
```diff
@@ -175,16 +175,48 @@ Quick Start
 <% if js? -%>
 ### Frontend
 
+The JavaScript API client gives you access to low level methods to search and
+receive results. This is all you need for building your front-end but will
+require a bit of custom code on your side for displaying the results. Reading
```
Comment (bobylito) [inline include/templates/README.md]: Let's be honest, it does require more than a bit of custom to do stuff with the client, we could remove the 'bit' part :)

File: include/templates/README.md [hunk 4]
```diff
@@ -175,16 +175,48 @@ Quick Start
...
+#### [instantsearch.js](https://community.algolia.com/instantsearch.js/)
+
+[instantsearch.js](https://community.algolia.com/instantsearch.js/) is for **full
+page** search.
+
+![instantsearch.js example](https://raw.githubusercontent.com/algolia/algoliasearch-client-js/master/examples/instantsearch.gif)
+
+We strongly encourage you to have a look at those libraries because they are
+packaged with a lot of options that will cover most of your needs without
+requiring you to do all the plumbing.
```
Comment (bobylito) [inline include/templates/README.md]: Maybe we could talk about the helper in this section too? Probably when it has its own website out though :P

**Comment by bobylito** [comment]
Left some comments, otherwise good to go :)

File: include/templates/README.md (hunk 4 above)
Comment (pixelastic) [inline include/templates/README.md]: I wasn't sure how to introduce the helper in here. I'd like people to use autocomplete and instantsearch if they have the need for such a UI, but we should really advertise the helper as well.

But yes, maybe wait until it has its own website.

**Pull request merged**